hal/api/Serial.h

```diff
@@ -74,6 +74,13 @@ class Serial : public SerialBase, public Stream {
     PlatformMutex _mutex;
 };
 
+/** Get the stdio Serial object, which is lazy instantiated. There's only 
+ *  one stdio object that should be used within an application.
+ *
+ *  @returns stdio object 
+ */
+Serial& get_stdio_serial();
```

Comment:
> With Serial& it's easy to accidentally do this:

That's a user error, we can protect it as @pan- highlighted it's about copy ctor. I'll have a look at it